src/validator.ts

```diff
@@ -1053,14 +1055,25 @@ function fnJoin(join: any, parts: any){
     return parts.join(join);
 }
 
-export function fnGetAtt(reference: string, attribute: string){
+export function fnGetAtt(reference: string, attributeName: string){
     if(workingInput['Resources'].hasOwnProperty(reference)){
         const resource = workingInput['Resources'][reference];
-        if (resource['Attributes'].hasOwnProperty(attribute)){
-            return resource['Attributes'][attribute];
-        } else if (resource['Type'].indexOf('Custom::') === 0 || resource['Type'] === 'AWS::CloudFormation::CustomResource') {
-            return `mockAttr_${reference}_${attribute}`;
-        } 
+        if (resource['Type'].indexOf('Custom::') === 0) {
+            return `mockAttr_${reference}_${attributeName}`;
+        } else if (resource['Type'] === 'AWS::CloudFormation::CustomResource') {
+            return `mockAttr_${reference}_${attributeName}`;
+        } else {
+            // Lookup attribute
+            const attribute = resourcesSpec.getResourceTypeAttribute(resource['Type'], attributeName)
```

Comment:
We need to catch the exception and throw it back to the user as a Critical error...

Also need a test case for this type of behaviour, for example requesting "!GetAtt DNSVPCDomain.VeryLostNameServers"

should result in:
Resource: Resources > .....
Message: Required property ...
Documentation: ...
Template invalid!


Such examples are
```
addError('crit', `Fn::If does not allow ${keys[0]} as a nested function within an array`, placeInTemplate, 'Fn::If');
```

Probably: Fn::GetAtt cannot find property AAAA on resource AAAA of type AAAA, placeInTemplate, DestinationType?? (if possible), otherwise fn::getatt

The last parameter decides which documentation to serve to the user.

There might be some better ones within the validator